File: 프로그래머스/Lv.0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        List<Integer> answer = new ArrayList<Integer>();
        for(int i =0;i <queries.length;i++){
            int num = 1000001;
            for(int j=queries[i][0]; j<= queries[i][1]; j++){
                if(arr[j] > queries[i][2]){
                    num = Math.min(arr[j], num);
                }
            }
            
            if(num == 1000001){
                answer.add(-1);    
            }else{
                answer.add(num);    
            }
            
        }
        return answer.stream().mapToInt(i->i).toArray();
    }
}

File: 프로그래머스/Lv.0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> queries) {
    vector<int> answer;
    for(int i=0;i<queries.size(); i++){
        int num = 1000001;
        for(int j=queries[i][0];j<=queries[i][1]; j++){
            if(arr[j] > queries[i][2])
                num = min(num, arr[j]);
        }
        if(num == 1000001) num = -1;
        answer.push_back(num);
    }
    return answer;
}